### PR TITLE
Bug 1678376 - Make t/001compile.t Great Again

### PR DESCRIPTION
**Pull request by dklawren**
Mainly this added some code to create a temp test DB using SQLite to allow the compiling of the Bugzilla/Model/* modules which require a DB. Also some critic warnings fixed in Archive.pm and 001compile.t

This does not *yet* test extension modules as there is added complexity to get them to load without lots of redefined warnings. I will address extensions as a separate bug.

### CODE REVIEW

**Review by albertoEnstein — APPROVED** [review]

File: t/001compile.t [hunk 1]
```diff
@@ -20,16 +20,20 @@ use Support::Files;
...
-    exit;
-  }
   plan tests => @Support::Files::testitems + @Support::Files::test_files;
 
   use_ok('Bugzilla::Constants');
   use_ok('Bugzilla::Install::Requirements');
   use_ok('Bugzilla');
+
+  if (-f 'data/db/model_test') {
+    unlink 'data/db/model_test' || die $!;
```
Comment (globau) [inline t/001compile.t:30]: This needs to be either `unlink('data/db/model_test') || die $!` or `unlink 'data/db/model_test' or die $!`

File: t/001compile.t [hunk 2]
```diff
@@ -45,28 +49,28 @@ sub compile_file {
...
 
-  if ($file =~ s/\.pm$//) {
+  if ($file =~ s/[.]pm$//) {
     $file =~ s{/}{::}g;
     use_ok($file);
     return;
   }
 
-  open(my $fh, $file);
+  open my $fh, '<', $file || die $!;
```
Comment (globau) [inline t/001compile.t:58]: Same `||` vs `or` bug

File: t/001compile.t [hunk 3]
```diff
@@ -45,28 +49,28 @@ sub compile_file {
...
   my $bang = <$fh>;
-  close $fh;
+  close $fh || die $!;
```
Comment (globau) [inline t/001compile.t:60]: Same `||` vs `or` bug

File: Bugzilla/Attachment/Archive.pm [hunk 4]
```diff
@@ -80,7 +80,7 @@ sub _build_input_fh {
     croak "I will not read and write a file at the same time";
   }
   my $file = $self->file;
-  return IO::File->new($self->file, '<:bytes') or die "cannot read $file: $!";
+  return IO::File->new($self->file, '<:bytes') || die "cannot read $file: $!";
```
Comment (globau) [inline Bugzilla/Attachment/Archive.pm:83]: This is unrelated and shouldn't be touched here.

**Review by globau — CHANGES_REQUESTED** [review]

File: Bugzilla/Attachment/Archive.pm (hunk 4 above)
Comment (dklawren) [reply Bugzilla/Attachment/Archive.pm:83]: When running t/001compile.t, I get the following error if this left as-is:

```
Possible precedence issue with control flow operator at Bugzilla/Attachment/Archive.pm line 83
```

**Review by globau — APPROVED** [review]
